### DISCUSSION TITLE
Safe to use pgsodium & vault in self-hosted project?

### DISCUSSION THREAD

**Discussion by riccardolardi** [supabase/supabase#29051] [Questions] [answered]
I'm working on a new project and would benefit from being able to use pgsodium & supabase vault to store some secrets and use their decrypted values in different SQL functions/procedures. Is the supabase vault going to be affected by the upcoming retirement of pgsodium by supabase? Or can we still safely use pgsodium through SQL to store and decrypt values in supabase vault?

**Comment by GaryAustin1** [comment, marked as answer]
Supabase has said they plan to keep Vault in some form.  They will deprecate pgsodium.  I don't know the details, but if you are self-hosted I would think you could install the extension on your own if/when it is no longer on supabase's release.
